### PR TITLE
Fixes in CalibTracker/SiStripChannelGain

### PR DESCRIPTION
**Pull request by Dr15Jones**
replaced non-standard __gnu_cxx::hash_map with std::unordered_map
removed the use of a static variable

### CODE REVIEW

**Comment by Dr15Jones** [comment]
Fixes problem in #18291

**Comment by Dr15Jones** [comment]
please test

File: CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc [hunk 1]
```diff
@@ -196,6 +193,7 @@ class SiStripGainFromCalibTree : public ConditionDBWriter<SiStripApvGain> {
         enum statistic_type {None=-1, StdBunch, StdBunch0T, FaABunch, FaABunch0T, IsoBunch, IsoBunch0T, Harvest};
 
         std::vector<string>    dqm_tag_;
+        std::vector booked_dir_;
```
Comment (ghellwig) [inline CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc]: @Dr15Jones what about using `std::string`? ;)

Comment (Dr15Jones) [inline CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc]: <head slap> helps when one compiles the code before doing the pull request</head slap>

**Comment by Dr15Jones** [comment]
Weird, I deleted the original branch from github that had the problem. I then pushed a new branch with the same name to github and told github to use that new branch for a pull request. However, github kept the code from the original branch and not the new branch!

File: CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc (hunk 1 above)
Comment (Dr15Jones) [inline CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc]: Turns out you can't reuse branch names in GitHub. After several tries I always get the code changes from the original and not the newest version.

**Comment by Dr15Jones** [comment]
Turns out, I just plain forgot to commit my change! Mondays, ick.

**Pull request closed without merging**